### ISSUE TITLE
Query loses its result type when using `setLockMode` or `setHint`

### ISSUE THREAD

**Issue by noemi-salaun** [phpstan/phpstan-doctrine#393]
When using `setLockMode()`  or `setHint()` on a `Query`, PHPStan stop inferring the result type.

## GOOD TYPE
```php
$user = $entityManager
    ->getRepository(User::class)
    ->createQueryBuilder('user')
    ->where('user.id = :userId')
    ->getQuery()
    ->getResult();

\PHPStan\dumpType($user); // -> Dumped type: array<App\User>
```

## MIXED TYPE
```php
$user = $entityManager
    ->getRepository(User::class)
    ->createQueryBuilder('user')
    ->where('user.id = :userId')
    ->getQuery()
    ->setLockMode(LockMode::PESSIMISTIC_READ) // <- Add this line
    ->getResult();

\PHPStan\dumpType($user); // -> Dumped type: mixed
```

**Comment by VincentLanglet** [comment]
Hi @noemi-salaun, this can be fixed directly on doctrine/orm.

The return type is `self` but the phpdoc should be more precise.
https://github.com/doctrine/orm/blob/2.13.x/lib/Doctrine/ORM/Query.php#L775
Adding `/** @return $this */` will solve the issue.

I opened the PR https://github.com/doctrine/orm/pull/10238

**Comment by VincentLanglet** [comment]
This will be fixed in 2.14 version of doctrine
